tests/unit/asio_service_stream_test.cxx

```diff
@@ -0,0 +1,227 @@
+
+        ptr<resp_msg> process_req(req_msg& req,
+                                const req_ext_params& ext_params) 
+        {
+            ptr<resp_msg> resp = cs_new<resp_msg>( state_->get_term(),
+                                        msg_type::append_entries_response,
+                                        id_,
+                                        req.get_src(),
+                                        log_store_->next_slot() );
+            if (req.get_last_log_idx() == streamed_log_index) {
+                streamed_log_index++;
+                resp->accept(streamed_log_index.load());
+                ptr<buffer> buf = req.log_entries().at(0)->get_buf_ptr();
+                buf->pos(0);
+                std::string buf_str = buf->get_str();
+                if (buf_str != test_msg) {
+                    SimpleLogger* ll = my_log_wrapper->getLogger();
+                    _log_info(ll, "resp str: %s", buf_str.c_str());
+                    msg_mismatch.store(true);
+                }
+            } else {
+                SimpleLogger* ll = my_log_wrapper->getLogger();
+                _log_info(ll, "req log index not match, req: %ld, current: %ld", 
+                req.get_last_log_idx(), streamed_log_index.load());
+            }
+            return resp;
+        }
+
+        ptr<logger_wrapper> my_log_wrapper;
+        std::atomic<ulong> streamed_log_index;
+        std::atomic<bool> msg_mismatch;
+    };
+
+    class stream_server {
+        public:
+        stream_server(int id, int port) : 
+        my_id(id),
+        port(port),
+        response_log_index(1)
+
+        {
+            init_server();
+        }
+
+        void send_req(int count) {
+            ptr<buffer> msg = buffer::alloc(test_msg.size() + 1);
+            msg->put(test_msg);
+
+            while(count > 0) {
+                ptr<req_msg> req ( cs_new<req_msg>
+                ( 1, msg_type::append_entries_request, 1, my_id,
+                1, send_log_index, 1 ) );
+
+                ptr<log_entry> log( cs_new<log_entry>
+                ( 0, msg, log_val_type::app_log ) );
+                req->log_entries().push_back(log);
+
+                rpc_handler h = (rpc_handler)std::bind
+                    ( &stream_server::handle_result,
+                        this,
+                        req,
+                        std::placeholders::_1,
+                        std::placeholders::_2 );
+                my_client->send(req, h);
+                send_log_index++;
+                count--;
+            }
+        }
+
+        void handle_result(ptr<req_msg>& req, ptr<resp_msg>& resp, ptr<rpc_exception>& err) {
+            if (resp->get_next_idx() == get_next_log_index()) {
+                response_log_index++;
+            } else {
+                SimpleLogger* ll = my_log_wrapper->getLogger();
+                _log_info(ll, "resp log index not match, resp: %ld, current: %ld", 
+                resp->get_next_idx(), get_next_log_index());
```

Comment:
This is the whole point of this test (checking whether request and response are delivered in order). Rather than leaving a log, there should be a boolean flag to set if failure happens, and check `CHK_FALSE` at the end.

Reply:
If entering into this branch, this `response_log_index` will not increase. So it will fail at `CHK_EQ(count, s.get_resp_log_index());`. Those logs is just for helping debug.